circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/ReplicationFactoryImpl.java

```diff
@@ -86,10 +86,10 @@ public Replication newInstance(TableReplication tableReplication) {
     Replication replication = null;
     if (partitionKeys == null || partitionKeys.isEmpty()) {
       replication = createUnpartitionedTableReplication(tableReplication, source, replica, sourceDatabaseName,
-          sourceTableName, replicaDatabaseName, replicaTableName, replicaTableLocation, replication);
+          sourceTableName, replicaDatabaseName, replicaTableName, replicaTableLocation);
```

Comment:
`replication` is initialised with `null` and then gets the value from `createUnpartitionedTableReplication` or `createPartitionedTableReplication`, which means that the value does not need to change by reference.